app/src/main/java/de/westnordost/streetcomplete/overlays/surface/PathSurfaceOverlay.kt

```diff
@@ -0,0 +1,210 @@
+    var noteProvided: String? = null
+    when (surfaceStatus) {
+        is SingleSurfaceWithNote -> {
+            dominatingSurface = surfaceStatus.surface
+            noteProvided = surfaceStatus.note
+        }
+        is CyclewayFootwaySurfacesWithNote -> if (surfaceStatus.cycleway in UNDERSPECIFED_SURFACES && surfaceStatus.cyclewayNote == null) {
+            // the worst case possible - bad surface without note: so lets present it
+            dominatingSurface = surfaceStatus.cycleway
+            noteProvided = surfaceStatus.cyclewayNote
+        } else if (surfaceStatus.footway in UNDERSPECIFED_SURFACES) {
+            // cycleway surface either has
+            // data as bad as this one (also bad surface, without note)
+            // or even worse (bad surface without note, while here maybe there is a note)
+            dominatingSurface = surfaceStatus.footway
+            noteProvided = surfaceStatus.footwayNote
+        } else if (surfaceStatus.cycleway in UNDERSPECIFED_SURFACES) {
+            // so footway has no bad surface, while cycleway has bad surface
+            // lets take worse one
+            dominatingSurface = surfaceStatus.cycleway
+            noteProvided = surfaceStatus.cyclewayNote
+        } else {
+            // cycleway is arbitrarily taken as dominating here
+            // though for bicycles surface is a bit more important
+            dominatingSurface = surfaceStatus.cycleway
+        }
+        is SingleSurface -> {
+            dominatingSurface = surfaceStatus.surface
+        }
+        is CyclewayFootwaySurfaces -> if (surfaceStatus.footway in UNDERSPECIFED_SURFACES) {
+            dominatingSurface = surfaceStatus.footway
+        } else {
+            // cycleway is arbitrarily taken as dominating here
+            // though for bicycles surface is a bit more important
+            dominatingSurface = surfaceStatus.cycleway
+        }
+        is SurfaceMissing -> {
+            // no action needed
+        }
+        is SurfaceMissingWithNote -> {
+            // no action needed
+        }
+    }
+    // not set but indoor or private -> do not highlight as missing
+    val isNotSet = dominatingSurface in UNDERSPECIFED_SURFACES
+    val isNotSetButThatsOkay = isNotSet && (isIndoor(element.tags) || isPrivateOnFoot(element)) || element.tags["leisure"] == "playground"
+
+    val color = if (isNotSetButThatsOkay) {
+        Color.INVISIBLE
+    } else if (isNotSet && noteProvided != null) {
+        Color.BLACK
+    } else {
+        dominatingSurface.color
+    }
+    return if (element.tags["area"] == "yes") PolygonStyle(color) else PolylineStyle(StrokeStyle(color))
+}
+
+private fun getStyleForSidewalkAsProperty(element: Element): PolylineStyle {
+    val sidewalkSides = createSidewalkSides(element.tags)
+    // not set but on road that usually has no sidewalk or it is private -> do not highlight as missing
+    if (sidewalkSides == null || isPrivateOnFoot(element)) {
```

Comment:
comment conflicts with actual check made

Reply:
oh thanks for catching this!